getting-started/kubernetes/trying-ebpf.md

```diff
@@ -0,0 +1,190 @@
+
+### Before you begin...
+
+In the tech preview release, eBPF mode has the following pre-requisites:
+
+- A v5.3+ Linux kernel, for the tech preview release, we have limited our testing to Ubuntu 19.10 so we strongly recommend that you start with that too.
+- An underlying network that doesn't require Calico to use an overlay.  The instructions below guide you through setting up a cluster in a single AWS subnet; an alternative would be to set up your cluster on-prem with a routed network topology.  Using the IPIP or VXLAN overlay modes add additional overhead making it harder to see the benefits of eBPF mode. 
+- The network must be configured to allow {{site.prodname}} packets between Calico hosts.  VXLAN is used to forward packets from outside the cluster that are destined for NodePorts and ClusterIPs.
+- Single-homed hosts; eBPF mode currently assumes a single "main" host IP.
+- Must use the Calico CNI plugin and Calico IPAM.  It is not compatible with third-party CNI plugins (AWS CNI/Azure CNI/GKE CNI/flannel etc).
+- IPv4 only.  The tech preview release does not support IPv6.
+- Kubernetes API Datastore only.
+- Typha is not supported in the tech preview. 
+- The base [requirements]({{site.baseurl}}/getting-started/kubernetes/requirements) also apply.
+
+### How to
+
+- [Set up a suitable cluster](#set-up-a-suitable-cluster)
+- [Install Calico on nodes](#install-calico-on-nodes)
+
+#### Set up a suitable cluster
+
+We recommend using `kubeadm` to bootstrap a suitable cluster on AWS.  
+
+1. In AWS create a controller node and at least 2 worker nodes in the same VPC subnet.  Use Ubuntu 19.10 as the image.
+
+1. On each node, {% include open-new-window.html text='install kubeadm' url='https://kubernetes.io/docs/setup/production-environment/tools/kubeadm/install-kubeadm/' %}; we recommend using docker as the containeriser in order to match our test environment as closely as possible.
+
+1. Create the controller node of a new cluster. On the controller VM, execute:
+
+   ```
+   sudo kubeadm init --pod-network-cidr 192.168.0.0/16
+   ```
+
+1. To set up kubectl for the ubuntu user, run:
+
+   ```
+   mkdir -p $HOME/.kube
+   sudo cp -i /etc/kubernetes/admin.conf $HOME/.kube/config
+   sudo chown $(id -u):$(id -g) $HOME/.kube/config
+   ```
+   The final line of the kubeadm init output contains the command for joining your workers to the controller.  Run this on each worker, prepending `sudo` to run it as root.  It will look something like this:
+
+   ```
+   sudo kubeadm join 10.240.0.11:6443 --token <token> --discovery-token-ca-cert-hash sha256:<hash>
+   ```
+
+1. On the controller, verify that all nodes have joined
+
+   ```
+   kubectl get nodes
+   ```
+   which should output something similar to:
+
+   ```
+   NAME         STATUS     ROLES    AGE     VERSION
+   controller   NotReady   master   5m49s   v1.17.2
```

Comment:
Nodenames on AWS tend to have the form `ip-xxx-xxx-xxx-xxx.<region>.compute-internal`.  e.g. 
```
$ kubectl get no
NAME                                           STATUS   ROLES    AGE   VERSION
ip-172-16-101-119.us-west-2.compute.internal   Ready    <none>   86m   v1.17.3
ip-172-16-101-27.us-west-2.compute.internal    Ready    <none>   86m   v1.17.3
ip-172-16-101-49.us-west-2.compute.internal    Ready    master   88m   v1.17.3
ip-172-16-102-174.us-west-2.compute.internal   Ready    <none>   86m   v1.17.3
```